traitlets/config/application.py

```diff
@@ -899,6 +899,8 @@ def _load_config_files(
 
         yield each config object in turn.
         """
+        if os.path.isabs(basefilename):
+            path = None
```

Comment:
Probably need to make sure this case is tested, because this will fail on line 906 with:

```
---->  reversed(path)

TypeError: 'NoneType' object is not reversible
```

I think maybe you want:

```suggestion
            path = [None]
```